### PR TITLE
ci-cd: Add section on connecting build slaves over SSH

### PR DESCRIPTION
**Pull request by jonte**

### CODE REVIEW

File: docs/chapters/infrastructure/ci-cd/build-slaves.rst [hunk 1]
```diff
@@ -16,20 +16,62 @@ In order to run a slave which the master can reach via SSH you still need to pre
...
+
+Add the public key file (``build_slave_key.pub``) to ``/home/jenkins/.ssh/authorized_keys`` on your Jenkins *build slave* (or replace ``jenkins`` with the user you chose for your Jenkins build slave). Save the private key. It will be used when configuring Jenkins.
+
+Configuring Jenkins
+"""""""""""""""""""
+
+1. Go to Jenkins -> Manage Jenkins -> Nodes -> New Node
+2. Add the node name and set "Permanent Agent"
+3. Put in the name and the remote root directory, you should have a special user like ``jenkins`` to run it so set ``/home/jenkins/``
+4. Set the lables which this slave can run, for example ``DockerCI``
```
Comment (caybro) [inline docs/chapters/infrastructure/ci-cd/build-slaves.rst]: s/lables/labels

File: docs/chapters/infrastructure/ci-cd/build-slaves.rst [hunk 2]
```diff
@@ -16,20 +16,62 @@ In order to run a slave which the master can reach via SSH you still need to pre
...
+    * 7a. Set "Domain" to "Global credentials (unrestricted)"
+    * 7b. Set "Kind" to "SSH username with private key"
+    * 7c. Set "Username" to "jenkins" (assuming that you use this UNIX username for your jenkins user)
+    * 7d. Set "Private Key" to "Enter Directly", and supply the *private* key you generated previously
+    * 7e. Set "Passphrase" as needed, or leave empty if no passphrase is set
+    * 7f. Set "Description" to a suitable description
+
+8. Select the correct key in "Credentials"
+9. Set "Host Key Verification Strategy" to "Non verifying Verification Strategy"
+10. Set "Availability" to Keep this agent online as much as possible"
```
Comment (caybro) [inline docs/chapters/infrastructure/ci-cd/build-slaves.rst]: Unmatched quote (") at the end?

**Review by caybro — CHANGES_REQUESTED** [review]
Some minor typos spotted

File: docs/chapters/infrastructure/ci-cd/build-slaves.rst [hunk 3]
```diff
@@ -16,20 +16,62 @@ In order to run a slave which the master can reach via SSH you still need to pre
    sudo useradd jenkins
    sudo usermod -aG docker jenkins
 
+Connecting to the slave using SSH
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Generating SSH keypairs
+"""""""""""""""""""""""
+
+A private & public SSH keypair must be generated. The Jenkins master server will connect to the Jenkins build slaves using SSH. The Jenkins master server will have the SSH *private* key, in order to initiate the SSH connection, and the Jenkins build slaves will have *public* keys added to their respective ``jenkins`` users.
```
Comment (breakreturn) [inline docs/chapters/infrastructure/ci-cd/build-slaves.rst]: Why do you use an '&' in this sentence? We generally just say 'and' right?

**Review by breakreturn — CHANGES_REQUESTED** [review]
Tiny style nitpick, otherwise nice

File: docs/chapters/infrastructure/ci-cd/build-slaves.rst (hunk 3 above)
Comment (jonte) [reply docs/chapters/infrastructure/ci-cd/build-slaves.rst]: Not sure. I guess I write like this sometimes.. I'll change it though!

**Comment by jonte** [comment]
Thanks for the reviews. I fixed the issues & am ready for merging!

**Review by caybro — APPROVED** [review]

**Review by breakreturn — APPROVED** [review]

**Pull request merged**